### PR TITLE
Egd 7948 extend log dumping

### PR DESCRIPTION
**Pull request by alekrudnik**
Log are now dumped:
    * on system shutdown including hard faults and aborts
    * every 5 minutes
    
    Fixed some logs

### CODE REVIEW

**Review by wojtekrzepecki — APPROVED** [review]

File: module-sys/SystemManager/SystemManagerCommon.cpp [hunk 1]
```diff
@@ -371,7 +398,7 @@ namespace sys
     bool SystemManagerCommon::DestroyApplication(const std::string &name, Service *caller)
     {
         cpp_freertos::LockGuard lck(appDestroyMutex);
-        if (RequestServiceClose(name, caller)) {
+        if (RequestServiceClose(name, caller, 1000)) {
```
Comment (tomaszkrosnowski) [inline module-sys/SystemManager/SystemManagerCommon.cpp:401]: This has been 5000 by default. Why shortening time so much?

File: module-vfs/logdump/logdump.cpp [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// Copyright (c) 2017-2021, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#include <logdump/logdump.h>
+#include <purefs/filesystem_paths.hpp>
+#include <Logger.hpp>
+
+int dumpLogs()
+{
+    const auto logPath = purefs::dir::getLogsPath() / LOG_FILE_NAME;
```
Comment (piotr-tanski) [inline module-vfs/logdump/logdump.cpp:10]: const lvalues are not moveable. Copy constructor will be called.

File: module-vfs/logdump/include/logdump/logdump.h [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+#pragma once
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+    /// @return: < 0 - error occurred during log flush
+    /// @return:   0 - log flush did not happen
+    /// @return:   1 - log flush successful
+    int dumpLogs();
```
Comment (piotr-tanski) [inline module-vfs/logdump/include/logdump/logdump.h:13]: Is it ok that log dump functionality is a part of module-vfs and not a logger itself (with needed dependencies injected)?
Actually, the only dependency needed here is a dump file path.

File: module-sys/SystemManager/SystemManagerCommon.cpp (hunk 1 above)
Comment (alekrudnik) [reply module-sys/SystemManager/SystemManagerCommon.cpp:401]: it was temporary change, will revert

Comment (alekrudnik) [reply module-sys/SystemManager/SystemManagerCommon.cpp:401]: tx

**Review by tomaszkrosnowski — APPROVED** [review]

File: module-sys/SystemManager/SystemManagerCommon.cpp (hunk 1 above)
Comment (piotr-tanski) [inline module-sys/SystemManager/SystemManagerCommon.cpp:401]: What's 1000? Why is it different for apps than other services?

File: module-sys/SystemManager/SystemManagerCommon.cpp [hunk 4]
```diff
@@ -101,6 +101,47 @@ namespace sys
         LOG_DEBUG("%s", (GetName() + ":destructor").c_str());
     }
 
+    void logPowerOffReason(SystemManagerCommon::State state)
```
Comment (piotr-tanski) [inline module-sys/SystemManager/SystemManagerCommon.cpp:104]: Why is it left out of any namespace qualification?

File: module-vfs/logdump/logdump.cpp (hunk 2 above)
Comment (alekrudnik) [reply module-vfs/logdump/logdump.cpp:10]: ctrl + c, ctrl + v.
Fixed

File: module-sys/SystemManager/SystemManagerCommon.cpp (hunk 1 above)
Comment (alekrudnik) [reply module-sys/SystemManager/SystemManagerCommon.cpp:401]: temporary change, reverted, tx

File: module-sys/SystemManager/SystemManagerCommon.cpp (hunk 4 above)
Comment (alekrudnik) [reply module-sys/SystemManager/SystemManagerCommon.cpp:104]: fixed

File: module-vfs/logdump/include/logdump/logdump.h (hunk 3 above)
Comment (alekrudnik) [reply module-vfs/logdump/include/logdump/logdump.h:13]: moved to logger

Comment (alekrudnik) [reply module-vfs/logdump/include/logdump/logdump.h:13]: still believe that neither logger nor vfs are proper places to put this. IMHO, it should be moved to some product/family of products common place. However, lets not do it now - there is more things to clean up

**Review by niepiekm — APPROVED** [review]

**Review by piotr-tanski — APPROVED** [review]

**Pull request merged**